cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/MasterEnvironmentSparkSubmitter.java

```diff
@@ -127,6 +127,15 @@ protected void triggerShutdown() {
     sparkExecutionService.stopAndWait();
   }
 
+  protected boolean hasSparkDriverSucceeded() {
```

Comment:
Add `@Override` annotation and `throws Exception` to the method.

Reply:
done